src/Sarif/DelegatingStream.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    internal class DelegatingStream : Stream
+    {
+        private readonly Stream stream;
+
+        /// <summary>
+        /// This is a wrapper for a stream that protects the underlying stream from being disposed.
+        /// </summary>
+        /// <param name="underlyingStream"></param>
+        internal DelegatingStream(Stream underlyingStream)
+        {
+            stream = underlyingStream;
+            stream.Seek(0L, SeekOrigin.Begin);
+        }
+        public override bool CanRead => stream.CanRead;
+        public override bool CanSeek => stream.CanSeek;
+        public override bool CanWrite => stream.CanWrite;
+        public override long Length => stream.Length;
+        public override long Position
+        {
+            set
+            {
+                stream.Position = value;
```

Comment:
>Position [](http://example.com/codeflow?start=23&length=8)

can't do this if the stream isn't seekable. there are other methods that will throw as well, SetLength, I think. check the online docs on CanSeek for more.

Reply:
What behavior would you prefer? This is essentially just allowing the exceptions being thrown by the underlying stream to surface.  The alternative would be to make a seekable wrapper around non seekable streams.

Reply:
@eddynaka can fix this, I say don't do anything, just make this wrapper a lightweight delegate. If someone tries to call methods illegally on a non-seekable stream they'll get the same behavior as if they called the underlying instance directly.

Reply:
done :)